### ISSUE TITLE
Can't assign cookies to elastic agents

### ISSUE THREAD

**Issue by ketan** [gocd/gocd#3380]
##### Issue Type

- Bug Report

##### Summary

The docker swarm plugin uses a uuid of the form `docker-swarm.91f5d053-2985-4af5-81b8-4a148ea62265`, and it appears that assigning a cookie is failing with some exceptions.

```
Caused by: org.springframework.dao.DataIntegrityViolationException: could not insert: [com.thoughtworks.go.server.domain.AgentCookie]; SQL [insert into agents (id, uuid, cookie) values (null, ?, ?)]; nested exception is org.hibernate.exception.DataException: could not insert: [com.thoughtworks.go.server.domain.AgentCookie]
	at org.springframework.orm.hibernate3.SessionFactoryUtils.convertHibernateAccessException(SessionFactoryUtils.java:648)
	at org.springframework.orm.hibernate3.HibernateAccessor.convertHibernateAccessException(HibernateAccessor.java:412)
	at org.springframework.orm.hibernate3.HibernateTemplate.doExecute(HibernateTemplate.java:411)
	at org.springframework.orm.hibernate3.HibernateTemplate.executeWithNativeSession(HibernateTemplate.java:374)
	at org.springframework.orm.hibernate3.HibernateTemplate.saveOrUpdate(HibernateTemplate.java:737)
	at com.thoughtworks.go.server.persistence.AgentDao$1.doInTransactionWithoutResult(AgentDao.java:81)
	at org.springframework.transaction.support.TransactionCallbackWithoutResult.doInTransaction(TransactionCallbackWithoutResult.java:33)
	at com.thoughtworks.go.server.transaction.TransactionTemplate$2.doInTransaction(TransactionTemplate.java:39)
	at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:130)
	at com.thoughtworks.go.server.transaction.TransactionTemplate.execute(TransactionTemplate.java:35)
	at com.thoughtworks.go.server.persistence.AgentDao.associateCookie(AgentDao.java:73)
	at com.thoughtworks.go.server.service.AgentService.assignCookie(AgentService.java:367)
	at com.thoughtworks.go.remote.BuildRepositoryRemoteImpl.getCookie(BuildRepositoryRemoteImpl.java:132)
	... 83 more
Caused by: org.hibernate.exception.DataException: could not insert: [com.thoughtworks.go.server.domain.AgentCookie]
	at org.hibernate.exception.SQLStateConverter.convert(SQLStateConverter.java:100)
	at org.hibernate.exception.JDBCExceptionHelper.convert(JDBCExceptionHelper.java:66)
	at org.hibernate.id.insert.AbstractReturningDelegate.performInsert(AbstractReturningDelegate.java:64)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2176)
	at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2656)
	at org.hibernate.action.EntityIdentityInsertAction.execute(EntityIdentityInsertAction.java:71)
	at org.hibernate.engine.ActionQueue.execute(ActionQueue.java:279)
	at org.hibernate.event.def.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:321)
	at org.hibernate.event.def.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:204)
	at org.hibernate.event.def.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:130)
	at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.saveWithGeneratedOrRequestedId(DefaultSaveOrUpdateEventListener.java:210)
	at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.entityIsTransient(DefaultSaveOrUpdateEventListener.java:195)
	at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.performSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:117)
	at org.hibernate.event.def.DefaultSaveOrUpdateEventListener.onSaveOrUpdate(DefaultSaveOrUpdateEventListener.java:93)
	at org.hibernate.impl.SessionImpl.fireSaveOrUpdate(SessionImpl.java:535)
	at org.hibernate.impl.SessionImpl.saveOrUpdate(SessionImpl.java:527)
	at org.hibernate.impl.SessionImpl.saveOrUpdate(SessionImpl.java:523)
	at org.springframework.orm.hibernate3.HibernateTemplate$16.doInHibernate(HibernateTemplate.java:740)
	at org.springframework.orm.hibernate3.HibernateTemplate.doExecute(HibernateTemplate.java:406)
	... 93 more
Caused by: org.h2.jdbc.JdbcSQLException: Value too long for column "UUID VARCHAR(40) NOT NULL SELECTIVITY 100": "'docker-swarm.45192de8-65b7-4af6-ab0e-24995b138ecd' (49)"; SQL statement:
insert into agents (id, uuid, cookie) values (null, ?, ?) [22001-168]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:329)
	at org.h2.message.DbException.get(DbException.java:169)
	at org.h2.table.Column.validateConvertUpdateSequence(Column.java:315)
	at org.h2.table.Table.validateConvertUpdateSequence(Table.java:689)
	at org.h2.command.dml.Insert.insertRows(Insert.java:120)
	at org.h2.command.dml.Insert.update(Insert.java:84)
	at org.h2.command.CommandContainer.update(CommandContainer.java:75)
	at org.h2.command.Command.executeUpdate(Command.java:230)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:156)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:142)
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
	at org.hibernate.id.IdentityGenerator$GetGeneratedKeysDelegate.executeAndExtract(IdentityGenerator.java:94)
	at org.hibernate.id.insert.AbstractReturningDelegate.performInsert(AbstractReturningDelegate.java:57)
	... 109 more
```

**Comment by varshavaradarajan** [comment]
The uuid being inserted is `docker-swarm.45192de8-65b7-4af6-ab0e-24995b138ecd` . Thats not really the uuid, its the hostname or the agent-id. Looks like we are passing incorrect values to the SQL.

**Comment by jyotisingh** [comment]
The `builds` table refers to the agent-uuid. The current column size wouldn't suffice. If we continue to add the uuid generated by the plugin, then the builds table will have to be updated too. https://github.com/gocd/gocd/pull/3382 doesn't take care of that.

**Comment by ketan** [comment]
Closed via #3382
